### PR TITLE
See #7389. One tick process per quorum queue

### PR DESCRIPTION
**Pull request by SimonUnge**
## Proposed Changes

Using the pid and aux state trick to check if the QQ already have a process running for a tick.

## Types of Changes

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [x] Bug fix (non-breaking change which fixes issue #7389)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause an observable behavior change in existing systems)
- [ ] Documentation improvements (corrections, new content, etc)
- [ ] Cosmetic change (whitespace, formatting, etc)
- [ ] Build system and/or CI

## Checklist

_Put an `x` in the boxes that apply.
You can also fill these out after creating the PR.
If you're unsure about any of them, don't hesitate to ask on the mailing list.
We're here to help!
This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the `CONTRIBUTING.md` document
- [x] I have signed the CA (see https://cla.pivotal.io/sign/rabbitmq)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] All tests pass locally with my changes
- [ ] If relevant, I have added necessary documentation to https://github.com/rabbitmq/rabbitmq-website
- [ ] If relevant, I have added this change to the first version(s) in release-notes that I expect to introduce it

### CODE REVIEW

File: deps/rabbit/src/rabbit_fifo.erl [hunk 1]
```diff
@@ -1030,6 +1029,16 @@ handle_aux(leader, cast, {#return{msg_ids = MsgIds,
         _ ->
             {no_reply, Aux0, Log0}
     end;
+handle_aux(leader, _, {handle_tick, Fun, Args},
+           #?AUX{tick_pid = undefined} = Aux, Log, _) ->
+    Pid = spawn(rabbit_quorum_queue, Fun, Args),
```
Comment (kjnilsson) [inline deps/rabbit/src/rabbit_fifo.erl:1034]: I think we can skip the monitor churn and just check the previous pid using `is_process_alive/1` on the next tick

**Review by kjnilsson — CHANGES_REQUESTED** [review]
This looks good but added a suggestion to avoid issuing a monitor effect for the tick process.

File: deps/rabbit/src/rabbit_fifo.erl (hunk 1 above)
Comment (SimonUnge) [reply deps/rabbit/src/rabbit_fifo.erl]: Done! @kjnilsson

Comment (SimonUnge) [reply deps/rabbit/src/rabbit_fifo.erl]: Actually, need to remove unnecessary function clause. Hang on....

Comment (SimonUnge) [reply deps/rabbit/src/rabbit_fifo.erl]: Aaaaand done.

**Comment by SimonUnge** [comment]
> This looks good but added a suggestion to avoid issuing a monitor effect for the tick process.

Fixed!

File: deps/rabbit/src/rabbit_fifo.erl [hunk 2]
```diff
@@ -894,10 +894,9 @@ tick(Ts, #?MODULE{cfg = #cfg{name = _Name,
                              resource = QName}} = State) ->
     case is_expired(Ts, State) of
         true ->
-            [{mod_call, rabbit_quorum_queue, spawn_deleter, [QName]}];
+            [{aux, {handle_tick, spawn_deleter, [QName]}}];
```
Comment (kjnilsson) [inline deps/rabbit/src/rabbit_fifo.erl:897]: I think we always want to launch this process irrespective if there is a tick process running.

File: deps/rabbit/src/rabbit_fifo.erl [hunk 3]
```diff
@@ -1030,6 +1029,18 @@ handle_aux(leader, cast, {#return{msg_ids = MsgIds,
         _ ->
             {no_reply, Aux0, Log0}
     end;
+handle_aux(leader, _, {handle_tick, Fun, Args},
+           #?AUX{tick_pid = Pid} = Aux, Log, _) ->
+    NewPid =
+        case Pid =:= undefined orelse is_process_alive(Pid) =:= false of
```
Comment (kjnilsson) [inline deps/rabbit/src/rabbit_fifo.erl:1035]: Suggestion, avoid inline boolean logic by introducing a new local function`process_is_alive/1` that handles the undefined case and spawn a process only if this function returns false.

**Review by kjnilsson — CHANGES_REQUESTED** [review]
nearly there! :)

File: deps/rabbit/src/rabbit_fifo.erl (hunk 3 above)
Comment (SimonUnge) [reply deps/rabbit/src/rabbit_fifo.erl:1035]: Got it, will do.

File: deps/rabbit/src/rabbit_fifo.erl (hunk 2 above)
Comment (SimonUnge) [reply deps/rabbit/src/rabbit_fifo.erl]: Done, I think.

**Comment by SimonUnge** [comment]
Done, part 2

**Comment by kjnilsson** [comment]
@SimonUnge approved, please can you squash all your commits into one?

**Comment by SimonUnge** [comment]
> @SimonUnge approved, please can you squash all your commits into one?

Done!

**Comment by SimonUnge** [comment]
@michaelklishin Should I rebase this on `main`?

**Comment by michaelklishin** [comment]
> @michaelklishin Should I rebase this on `main`?

I used Rebase and merge on GH. Thank you for your ongoing contributions!

**Comment by kjnilsson** [comment]
@Mergify backport v3.11.x

**Pull request merged**